rust/crd/src/authorization.rs

```diff
@@ -23,89 +50,107 @@ pub struct UserPermission {
 }
 
 #[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
 pub struct AccessPermission {
     pub read: Option<bool>,
     pub write: Option<bool>,
 }
 
-pub async fn create_or_update_rego_rule_resource(
+pub async fn create_rego_rules(client: &Client, trino: &TrinoCluster) -> Result<()> {
+    let spec: &TrinoClusterSpec = &trino.spec;
+
+    if let Some(authorization) = &spec.authorization {
+        let rego_rules = build_rego_rules(authorization)?;
+        create_or_update_rego_rule_resource(client, &authorization.package, rego_rules).await?;
+    }
+
+    Ok(())
+}
+
+async fn create_or_update_rego_rule_resource(
     client: &Client,
     package_name: &str,
     rego_rules: String,
-) -> OperatorResult<RegoRule> {
+) -> Result<RegoRule> {
     let new_rego_rule_spec = RegoRuleSpec { rego: rego_rules };
+    let namespace = "default";
 
     let rego_rule_resource = RegoRule {
-        api_version: "opa.stackable.tech/v1alpha1".to_string(),
-        kind: "RegoRule".to_string(),
         metadata: ObjectMeta {
             name: Some(package_name.to_string()),
             ..Default::default()
         },
         spec: new_rego_rule_spec.clone(),
     };
 
-    match client.get::<RegoRule>(package_name, Some("default")).await {
+    // TODO: make namespace configurable
```

Comment:
Resolve this TODO

Reply:
honestly no idea on how to do that... need to investigate

Reply:
Can't you just add it as an optional field to the `authorization` section in the CRD and then read it from there?

Reply:
I think the safe approach for now would be to just copy the namespace from the `TrinoCluster`. Cross-namespace access (especially cross-namespace writes) should probably be avoided since they break both ownership references and multitenancy expectations.

Reply:
yeah that makes sense, done